bob/src/api/mod.rs

```diff
@@ -143,16 +145,39 @@ pub(crate) struct SpaceInfo {
     occupied_disk_space_by_disk: HashMap<String, u64>
 }
 
-pub(crate) fn spawn<A>(bob: BobServer<A>, address: IpAddr, port: u16)
+async fn tls_server(tls_config: &Option<TLSConfig>, addr: SocketAddr) -> Option<AxumServer<RustlsAcceptor>> {
+    if let Some(tls_config) = tls_config {
+        if let Some(true) = tls_config.rest {
+            if let (Some(cert_path), Some(pkey_path)) = (&tls_config.cert_path, &tls_config.pkey_path) {
+                let config = RustlsConfig::from_pem_file(
+                    cert_path,
+                    pkey_path,
+                ).await.unwrap();
+                return Some(bind_rustls(addr, config));
+            } else {
+                error!("rest tls enabled, but certificate or private key not specified");
+                panic!("rest tls enabled, but certificate or private key not specified");
+            }
+        }
+    }
+    None
+}
+
+pub(crate) async fn spawn<A>(bob: BobServer<A>, tls_config: &Option<TLSConfig>, address: IpAddr, port: u16)
 where
     A: Authenticator,
 {
     let socket_addr = SocketAddr::new(address, port);
 
     let router = router::<A>().layer(Extension(bob));
-    let task = Server::bind(&socket_addr).serve(router.into_make_service());
-
-    tokio::spawn(task);
+    
+    if let Some(tls_server) = tls_server(tls_config, socket_addr).await {
```

Comment:
From architecture point of view it is better to check that TLS enabled on this level. `tls_server` function should assume, that it is enabled and receive `TLSConfig` as parameter (not `Option<TLSConfig>`)